packages/mgt-components/src/components/mgt-search-results/mgt-search-results.ts

```diff
@@ -0,0 +1,1268 @@
+        default:
+          return this.renderDefault(result);
+      }
+    }
+  }
+
+  /**
+   * Renders the footer with pages if required
+   *
+   * @param hitsContainer Search results
+   */
+  private renderFooter(hitsContainer: SearchHitsContainer) {
+    if (this.pagingRequired(hitsContainer)) {
+      const pages = this.getActivePages(hitsContainer.total);
+
+      return html`
+        <div class="search-results-pages">
+          ${this.renderPreviousPage()}
+          ${this.renderFirstPage(pages)}
+          ${this.renderAllPages(pages)}                
+          ${this.renderNextPage()}
+        </div>
+      `;
+    }
+  }
+
+  /**
+   * Validates if paging is required based on the provided results
+   *
+   * @param hitsContainer
+   */
+  private pagingRequired(hitsContainer: SearchHitsContainer) {
+    return hitsContainer?.moreResultsAvailable || this.currentPage * this.size < hitsContainer?.total;
+  }
+
+  /**
+   * Gets a list of active pages to render for paging purposes
+   *
+   * @param totalResults Total number of results of the search query
+   */
+  private getActivePages(totalResults: number) {
+    const getFirstPage = () => {
+      const medianPage = this.currentPage - Math.floor(this.pagingMax / 2) - 1;
+
+      if (medianPage >= Math.floor(this.pagingMax / 2)) {
+        return medianPage;
+      } else {
+        return 0;
+      }
+    };
+
+    const pages: number[] = [];
+    const firstPage = getFirstPage();
+
+    if (firstPage + 1 > this.pagingMax - this.currentPage || this.pagingMax === this.currentPage) {
+      for (
+        let i = firstPage + 1;
+        i < Math.ceil(totalResults / this.size) &&
+        i < this.pagingMax + (this.currentPage - 1) &&
+        pages.length < this.pagingMax - 2;
+        ++i
+      ) {
+        pages.push(i + 1);
+      }
+    } else {
+      for (let i = firstPage; i < this.pagingMax; ++i) {
+        pages.push(i + 1);
+      }
+    }
+
+    return pages;
+  }
+
+  /**
+   * Renders all sequential pages buttons
+   *
+   * @param pages
+   */
+  private renderAllPages(pages: number[]) {
+    return html`
+      ${pages.map(
+        page =>
+          html`
+            <fluent-button 
+              title="${strings.page} ${page}"
+              appearance="stealth" 
+              class="${page === this.currentPage ? 'search-results-page-active' : 'search-results-page'}" 
+              @click="${() => this.onPageClick(page)}">
+                ${page}
+            </fluent-button>`
+      )}`;
+  }
+
+  /**
+   * Renders the "First page" button
+   *
+   * @param pages
+   */
+  private renderFirstPage(pages: number[]) {
+    return html`
+      ${
+        pages.some(page => page === 1)
+          ? nothing
+          : html`
+              <fluent-button 
+                title="${strings.page} 1"
+                appearance="stealth" class="search-results-page" @click="${() =>
+                  this.onFirstPageClick()}">1</fluent-button>
```

Comment:
Let's make this more readable:
```suggestion
              <fluent-button 
                 title="${strings.page} 1"
                 appearance="stealth"
                 class="search-results-page"
                 @click="${this.onFirstPageClick}"
               >
                 1
               </fluent-button>
```